plugins/factory-plugin/src/zip.ts

```diff
@@ -0,0 +1,70 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+const { spawn } = require('child_process');
+const path = require('path');
+const fs = require('fs');
+
+export async function execZip(folder: string, zipfile: string): Promise<void> {
```

Comment:
As far as I remember, we already have existing generic code to invoke spawn (we use it for git) so maybe you could reuse it here.

Reply:
I guess you're talking about generator: 
[generator/src/command.ts](https://github.com/eclipse/che-theia/blob/50767320ae233bad5109d308e9b87d52b97e8ea1/generator/src/command.ts)

I don't see such generic code in `factory-plugin`.

Reply:
I was talking about 
https://github.com/eclipse/che-theia/blob/master/plugins/factory-plugin/src/git.ts#L57
but looks like it is not as generic as it could be ...

Reply:
but you could try to make it a bit more tiny, you have 2 commands that return the quite the same error message and do quite the same thing

Reply:
OK, I will refactor this part, 
thanks for review!